src/Microsoft.ML.TimeSeries/SsaChangePointDetector.cs

```diff
@@ -93,8 +104,21 @@ private static VersionInfo GetVersionInfo()
                 loaderAssemblyName: typeof(SsaChangePointDetector).Assembly.FullName);
         }
 
-        public SsaChangePointDetector(IHostEnvironment env, Arguments args, IDataView input)
-            : base(new BaseArguments(args), LoaderSignature, env, input)
+        // Factory method for SignatureDataTransform.
+        private static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(args, nameof(args));
+            env.CheckValue(input, nameof(input));
+
+            var transformer = new SsaChangePointDetector(env, args);
```

Comment:
>SsaChangePointDetector [](start = 34, length = 22)

as discussed during the brownbag today, 
1) trainable transformers should not have a public ctor. 
2) Add the 'training ctor' that takes `data` and trains.

This way, your transformer is stateful, and not fully initialized upon creation, we avoid this in our codebase.

Reply:
Also, making 'Model' public is premature.

---
In reply to: [227616404](https://github.com/dotnet/machinelearning/pull/1254#discussion_r227616404) [](ancestors = 227616404)